src/unix_socket_address.rs

```diff
@@ -0,0 +1,50 @@
+use std::ptr;
+
+use glib;
+use glib::object::Downcast;
+use glib::object::IsA;
+use glib::translate::*;
+
+use ffi;
+use SocketAddress;
+use UnixSocketAddress;
+use UnixSocketAddressType;
+
+use self::AddressType::*;
+
+pub enum AddressType<'a> {
+    Path(&'a str),
+    Anonymous,
+    Abstract(&'a str),
+    AbstractPadded(&'a str),
+}
+
+impl<'a> AddressType<'a> {
+    fn to_type(&self) -> UnixSocketAddressType {
+        match *self {
+            Path(_) => UnixSocketAddressType::Path,
+            Anonymous => UnixSocketAddressType::Anonymous,
+            Abstract(_) => UnixSocketAddressType::Abstract,
+            AbstractPadded(_) => UnixSocketAddressType::AbstractPadded,
+        }
+    }
+}
+
+pub trait UnixSocketAddressExtManual {
+    fn new_with_type(address_type: AddressType) -> UnixSocketAddress;
```

Comment:
This is a constructor, not a method. So should be simply in a `impl UnixSocketAddress` block, no new trait needed